src/api/ruleMapping/ruleMapping.service.spec.ts

```diff
@@ -84,15 +84,171 @@ describe('RuleMappingService', () => {
       });
     });
 
-    it('should handle empty nodes array', () => {
+    it('should handle empty nodes array', async () => {
       const nodes: Node[] = [];
-      const result = service.extractFields(nodes, 'outputs');
+      const result = await service.extractFields(nodes, 'outputs');
       expect(result).toEqual({ outputs: [] });
     });
+    it('should handle decisionNode correctly', async () => {
+      const nodes: Node[] = [
+        {
+          type: 'decisionNode',
+          content: {
+            key: 'someKey',
+          },
+          id: 'testNode',
+        },
+      ];
+
+      // Mock the ruleSchemaFile method to return a sample schema
+      jest.spyOn(service, 'ruleSchemaFile').mockResolvedValue({
+        inputs: [{ id: '1', name: 'Input1', type: 'string', property: 'field1' }],
+        resultOutputs: [{ id: '2', name: 'Output1', type: 'number', property: 'field2' }],
+      });
+
+      const result = await service.extractFields(nodes, 'inputs');
+      expect(result).toEqual({
+        inputs: [{ id: '1', name: 'Input1', type: 'string', property: 'field1' }],
+      });
+
+      const resultOutputs = await service.extractFields(nodes, 'outputs');
+      expect(resultOutputs).toEqual({
+        outputs: [{ id: '2', name: 'Output1', type: 'number', property: 'field2' }],
+      });
+    });
+
+    it('should handle expressionNode with simple expressions correctly', async () => {
+      const nodes: Node[] = [
+        {
+          type: 'expressionNode',
+          content: {
+            expressions: [
+              { key: 'expr1', value: 'field3' },
+              { key: 'expr2', value: '123' },
+            ],
+          },
+          id: 'testNode',
+        },
+      ];
+
+      const result = await service.extractFields(nodes, 'inputs');
+      expect(result).toEqual({
+        inputs: [
+          { key: 'expr1', property: 'field3', exception: null },
+          { key: 'expr2', property: '123', exception: null },
+        ],
+      });
+    });
+
+    it('should handle expressionNode with complex expressions correctly', async () => {
+      const nodes: Node[] = [
+        {
+          type: 'expressionNode',
+          content: {
+            expressions: [
+              { key: 'expr1', value: 'field3' },
+              { key: 'expr2', value: 'complexExpr + 2' },
```

Comment:
First, thanks for these tests as I was wondering what the difference between a simple vs complex expression looked like. Second, I think the first one here is still a simple one though, right?

Reply:
Updated so they are both complex. Initially had only one as then you could compare in the same test and see that it worked correctly while not impacting non-complex expressions, but it's sort of redundant with the simple expressions test so revised per your comment.